content/04.workflows-and-software.md

```diff
@@ -80,36 +79,52 @@ Analysis workflows commonly rely on multiple software packages to generate final
+These features have led to widespread adoption within the bioinformatic community: packages for new software become quickly available, and can be installed easily across platforms.
+However, conda does not attempt to isolate software installations or standardize the runtime environment, meaning installations will not be completely reproducible over time.
+Heavyweight software management systems package not only the software of interest, but also the runtime environment information, with the goal of ensuring perfect reproducibility in software installation over time.
+Tools such as singularity and docker [@doi:10.1016/j.cels.2018.03.014; @doi:10.1038/s41592-018-0046-7; @doi:10.1371/journal.pone.0177459; @url:https://dl.acm.org/doi/10.5555/2600239.2600241] wrap software environments in "containers" that capture and reproduce the runtime environment information.
+Container-based management is particularly useful for systems where some dependencies may not be installable by lightweight managers. However, software installation within these containers can be limited by similar reproducibility issues, including changes in dependency installations over time.
+"Functional package managers" such as GNU Guix and Nix strictly require all dependency and configuration details be encoded within each software package, providing the most comprehensively reproducible installations.
+These have begun to be integrated into some bioinformatic tools, but have a steeper learning curve for independent use [@doi:10.1093/gigascience/giy123].
 
 
 ![**The conda package and environment manager simplifies software installation and management.**
-**A. Conda Recipe Repositories:** Each program distributed via Conda has a "recipe" describing all software dependencies needed for installation using Conda (each of which must also be installable via Conda). Recipes are stored and managed in the cloud in separate "channels", some of which specialize in particular fields or languages (e.g. the "bioconda" channel specializes in bioinformatic software, the "r" channel specializes in R language packages) [@doi:10.1038/s41592-018-0046-7]. **B. Use Conda Environments to Avoid Installation Conflicts:**  Conda does not require root
+**A. Conda Recipe Repositories:** Each program distributed via Conda has a "recipe" describing all software dependencies needed for installation using Conda (each of which must also be installable via Conda). Recipes are stored and managed in the cloud in separate "channels", some of which specialize in particular fields or languages (e.g. the "bioconda" channel specializes in bioinformatic software) [@doi:10.1038/s41592-018-0046-7]. **B. Use Conda Environments to Avoid Installation Conflicts:**  Conda does not require root
```

Comment:
```suggestion
**A. Conda Recipe Repositories:** Each program distributed via Conda has a "recipe" describing all software dependencies needed for installation using Conda (each of which must also be installable via Conda). Recipes are stored and managed in the cloud in separate "channels", some of which specialize in particular fields or languages (e.g. the "bioconda" channel specializes in bioinformatic software, while the "conda-forge" channel is a more general effort to provide and maintain standardized conda packages for a wide range of software) [@doi:10.1038/s41592-018-0046-7]. **B. Use Conda Environments to Avoid Installation Conflicts:**  Conda does not require root
```